### PR TITLE
Feature/improve simo caps negotiation

### PR DESCRIPTION
**Pull request by michaelgruner**

### CODE REVIEW

File: ext/tiovx/gsttiovxmultiscaler.c [hunk 1]
```diff
@@ -542,45 +547,217 @@ gst_tiovx_multi_scaler_create_graph (GstTIOVXSimo * simo, vx_context context,
...
+gst_tivox_multi_scaler_compute_named (GstTIOVXSimo * self,
+    GstStructure * structure, const gchar * name, GstTIOVXDimFunc func)
+{
+  const GValue *input = NULL;
+  GValue output = G_VALUE_INIT;
+
+  g_return_if_fail (self);
+  g_return_if_fail (structure);
+  g_return_if_fail (name);
+
```
Comment (emmadrigal) [inline ext/tiovx/gsttiovxmultiscaler.c:673]: Isn't `func` also a pointer? It might need a pointer guard

File: gst-libs/gst/tiovx/gsttiovxsimo.c [hunk 2]
```diff
@@ -769,35 +778,57 @@ gst_tiovx_simo_release_pad (GstElement * element, GstPad * pad)
...
-  GstCaps *peer_caps = NULL;
+
   GList *src_caps_list = NULL;
   GList *node = NULL;
 
   g_return_val_if_fail (self, NULL);
 
   priv = gst_tiovx_simo_get_instance_private (self);
 
+  GST_DEBUG_OBJECT (self, "Genereting src caps list");
```
Comment (emmadrigal) [inline gst-libs/gst/tiovx/gsttiovxsimo.c:810]: Typo `Generating`

File: gst-libs/gst/tiovx/gsttiovxsimo.h [hunk 3]
```diff
@@ -117,7 +117,9 @@ struct _GstTIOVXSimoClass
 
   gboolean      (*deinit_module)            (GstTIOVXSimo *self);
 
-  GstCaps *     (*get_caps)                 (GstTIOVXSimo *self, GstCaps *filter, GList *src_caps_list);
+  GstCaps *     (*get_sink_caps)            (GstTIOVXSimo *self, GstCaps *filter, GList *src_caps_list);
+
+  GstCaps *     (*get_src_caps)             (GstTIOVXSimo *self, GstCaps *filter, GstCaps *sink_caps);
```
Comment (emmadrigal) [inline gst-libs/gst/tiovx/gsttiovxsimo.h:122]: Update the documentation for this class

**Review by emmadrigal — CHANGES_REQUESTED** [review]

File: ext/tiovx/gsttiovxmultiscaler.c (hunk 1 above)
Comment (michaelgruner) [reply ext/tiovx/gsttiovxmultiscaler.c:673]: right.

**Pull request merged**